### PR TITLE
fix find_package() & add option to disable asan

### PR DESCRIPTION
**Pull request by poor-circle**
## What is changing

1. the target should add compile options `-fcoroutines`, so that user don't need to add it manually.
2. add option to disable asan.

### CODE REVIEW

File: async_simple/CMakeLists.txt [hunk 1]
```diff
@@ -23,6 +23,9 @@ if(UTHREAD)
 endif()
 
 # If there is no Uthread, async_simple is a header only library 
+if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  target_compile_options(async_simple_header_only INTERFACE "-fcoroutines")
```
Comment (ChuanqiXu9) [inline async_simple/CMakeLists.txt:27]: GCC enabled coroutines by default after 11. We need to check the version here. A message may be helpful here too.

File: async_simple/CMakeLists.txt [hunk 2]
```diff
@@ -22,6 +22,12 @@ if(UTHREAD)
   list(APPEND SRCS ${uthread_asm_src})
 endif()
 
+
+if (NOT (CMAKE_CXX_COMPILER_ID STREQUAL "Clang") AND (CMAKE_CXX_COMPILER_ID STREQUAL "GNU"))
```
Comment (ChuanqiXu9) [inline async_simple/CMakeLists.txt:26]:
```suggestion
if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
```

File: async_simple/CMakeLists.txt [hunk 3]
```diff
@@ -22,6 +22,12 @@ if(UTHREAD)
...
+  if (NOT CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "11.0.0" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "10.0.0")
```
Comment (ChuanqiXu9) [inline async_simple/CMakeLists.txt:27]:
```suggestion
  if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS "11.0.0")
```

File: async_simple/CMakeLists.txt (hunk 1 above)
Comment (poor-circle) [reply async_simple/CMakeLists.txt]: add check for gcc and gcc version

**Comment by poor-circle** [comment]
We need release a new version 1.3, then i will add async_simple to vcpkg.

**Review by ChuanqiXu9 — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**